build.gradle

```diff
@@ -202,6 +202,22 @@ model {
     }
 }
 
+project(':devApp') {
+    apply plugin: 'java'
+    apply plugin: 'application'
+
+    dependencies {
+        compile project(':')
+        runtime files(project(':').nativeTestFilesJar.archivePath)
+    }
+
+    sourceSets.main.java.srcDirs = [ "${rootDir}/src/dev/java" ]
```

Comment:
Id prefer uniformity over perfect, since it's either going to make the c++ devs unhappy or the java devs unhappy. And for these libs, since CPP is their main component I think I'd prefer keeping with cpp. I get what you mean by it being hacky, but we are just not doing something supported anyway. I would love if we had a build system that was perfect, but nothing is. Making the repo look like crap with code all over the place just to do a Java test app the Gradle way seems dumb to me.